### PR TITLE
Feature scan cleanup

### PR DESCRIPTION
**Pull request by matt-fevold**
2.1.24
Features:

N/A

Enhancements:

- Rewrites webbreaker webinspect scan code into a more testable code base.
  - Tests still need to be written, but now instead of a monolithic scan function with some weird dependencies there is a good number of small dedicated functions that are easy to single out.

Bugfixes:
N/A

### CODE REVIEW

File: webbreaker/webinspect/scan.py [hunk 1]
```diff
@@ -173,6 +222,440 @@ def _termination_event_handler(self):
...
+            config_helper = Config()
+            etc_dir = config_helper.etc
+            git_dir = os.path.join(config_helper.git, '.git')
+            try:
+                if self.scan_overrides.settings == 'Default':
+                    Logger.app.debug("Default settings were used")
+
+                    if os.path.isfile(self.scan_overrides.webinspect_upload_settings + '.xml'):
+                        self.scan_overrides.webinspect_upload_settings = self.scan_overrides.webinspect_upload_settings + '.xml'
+                    # if os.path.isfile(self.scan_overrides.webinspect_upload_settings):
```
Comment (HomerGaidarski) [inline webbreaker/webinspect/scan.py]: delete commented code

File: webbreaker/webinspect/webinspect_config.py [hunk 2]
```diff
@@ -108,250 +96,6 @@ def _get_webinspect_settings(self):
...
-            webinspectloghelper.log_git_access_error(self.webinspect_git, e)
-            raise Exception(webinspectloghelper.log_error_fetch_webinspect_configs())
-        except IndexError as e:
-            webinspectloghelper.log_config_file_unavailable(e)
-            raise Exception(webinspectloghelper.log_error_fetch_webinspect_configs())
-
-        Logger.app.debug("Completed webinspect config fetch")
-
     @staticmethod
     def _convert_verify_ssl_config(verify_ssl):
```
Comment (HomerGaidarski) [inline webbreaker/webinspect/webinspect_config.py]: this is the method i was talking about earlier, could you delete this and just use the webbreaker config method that does the same thing?

File: webbreaker/webinspect/common/loghelper.py [hunk 3]
```diff
@@ -47,4 +47,10 @@ def log_error_incorrect_webinspect_configs(self, args):
         Logger.app.error("Incorrect webinspect config values: {}".format(args))
 
     def log_error_fetch_webinspect_configs(self):
-        Logger.app.error("Fetching webinspect config")
\ No newline at end of file
+        Logger.app.error("Fetching webinspect config")
+
+    def log_error_invalid_ssl(self):
```
[the remark this answers is not in the document]
Comment (matt-fevold) [reply webbreaker/webinspect/common/loghelper.py]: Good catch 👍

**Comment by matt-fevold** [comment]
@HomerGaidarski I have remediated what you found. Thanks!

**Pull request merged**